docs/features/navigation.md

```diff
@@ -0,0 +1,121 @@
+        'title'      => 'Example Nested Category',
+        'capability' => 'view_woocommerce_reports',
+    )
+);
+```
+
+### Adding a menu item
+
+Adding an item, much like a category, can be added directly to the menu or to an existing category.  Typically this will create a link to the specified URL or callback unless overridden by JavaScript using the slot/fill approach described below.
+
+* `id` - (string) The unique ID of the menu item. Required.
+* `title` - (string) Title of the menu item. Required.
+* `parent` - (string) Parent menu item ID.
+* `capability` - (string) Capability to view this menu item.
+* `url` - (string) URL or callback to be used. Required.
+* `migrate` - (bool) Whether or not to hide the item in the wp admin menu.
+
+```php
+\Automattic\WooCommerce\Admin\Features\Navigation\Menu::add_plugin_item(
+    array(
+        'id'         => 'example-plugin',
+        'title'      => 'Example Plugin',
+        'capability' => 'view_woocommerce_reports',
+        'url'        => 'https://www.google.com',
+    )
+);
+```
+
+### Registering plugin screens
+
+In order to show the new navigation in place of the traditional WordPress menu on a given page, the screen ID must be registered to identify a page as supporting the new WooCommerce navigation.
+
+When adding items, the navigation will automatically add support for the screen via the URL or callback provided with an item.  However, custom post types need to be registered with the navigation to work on the custom post type page.
+
+
+```php
+\Automattic\WooCommerce\Admin\Features\Navigation\Screen::register_post_type( 'my-custom-post-type' );
+```
+
+You can also manually add a screen without registering an item.
+
+```php
+\Automattic\WooCommerce\Admin\Features\Navigation\Screen::add_screen( 'my-plugin-page' );
+```
+
+### Slot/fill items
+
+Using slot fill we can update items on the front-end of the site using JavaScript.  This is useful for modern JavaScript routing, more intricate interactions with menu items, or updating URLs and hyperlink text without reloading the page.
+
+In order to use slot fill, you can import the `WooNavigationItem` component from `@woocommerce/navigation` and match the `item` prop with the ID of the item you'd like to modify the behavior of.
+
+
+```js
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { registerPlugin } from "@wordpress/plugins";
+import { useHistory } from "react-router-dom";
+import { WooNavigationItem } from "@woocommerce/navigation";
+
+const MyPlugin = () => {
+    const history = useHistory();
+
+    const handleClick = () => {
+        history.push( '/my-plugin-path' );
+    }
+
+    return (
+        <WooNavigationItem item="example-plugin">
+            <Button onClick={ handleClick }>
+                { __( 'My Link', 'plugin-domain' ) }
+            </Button>
+        </WooNavigationItem>
+    );
+};
+
+registerPlugin('my-plugin', { render: MyPlugin });
+```
```

Comment:
One gotcha I ran into when setting this up from scratch with my example plugin is that when registering scripts, a dependency on `WC_ADMIN_APP` was required when registering js with `wp_register_script` when the slotFill was being used.

https://github.com/woocommerce/woocommerce-admin/blob/d5a0b32126f655bf3703aac21826397bf5a0df40/src/FeaturePlugin.php#L149

Something to that effect could save some headaches for developers. Separately, it would be great to build this into Woo's Dependency Extraction Webpack Plugin: https://github.com/woocommerce/woocommerce-admin/issues/5559

Reply:
I'm not quite seeing how to replicate this, but I believe I understand the issue.  Let's follow up in https://github.com/woocommerce/woocommerce-admin/issues/5559 and we can append info the docs if necessary.

Reply:
Sounds good. Right now, if you try to use slotFill in your app it will cause a fatal error. The reason is because wc-admin needs to load first, so 'wc-admin-app' needs to be added as a dependency. Its fine in the example extensions because thats technically part of wc-admin so load order doesn't matter.